components/Workspaces/TrackedRepoWizard/SearchByReposStep.tsx

```diff
@@ -34,52 +28,79 @@ const EmptyState = () => {
 };
 
 export const SearchByReposStep = ({
-  onAddToTrackingList,
-  onCancel,
   onSearch,
-  trackedReposCount,
+  onSelectRepo,
   repositories,
-  suggestedRepos,
+  searchedRepos,
+  suggestedRepos = [],
 }: SearchByReposStepProps) => {
-  // TODO: Implement these functions
-  const onFilterRepos = () => {};
-  const onSelectRepo = () => {};
+  const [filteredRepositories, setFilteredRepositories] = useState(repositories);
+
+  const onFilterRepos = (search: string) => {
+    setFilteredRepositories(
+      repositories.filter((repo) => {
+        return repo.includes(search);
+      })
+    );
+  };
+
+  const formRef = useRef<HTMLFormElement>(null);
+  const [searchIsLoading, setSearchIsLoading] = useState(false);
+
+  useEffectOnce(() => {
+    (formRef.current?.querySelector('[name="query"]') as HTMLInputElement)?.focus();
```

Comment:
gives focus to the search bar when you're on the search repos step of the tracked repo wizard.